src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -378,6 +399,26 @@ private HostSecrets GenerateHostSecrets()
             return _repository.WriteAsync(secrets.SecretsType, keyScope, secretsContent);
         }
 
+        private async Task PersistNonDecryptedSecrets<T>(T secrets, string keyScope = null) where T : ScriptSecrets
```

Comment:
Ideally we'd refactor `PersistSecretsAsync` to handle this as opposed to adding a new method that duplicates the persistence/serialization logic. At the very least, we can remove the duplication and call the existing method at the end.